spec/message.abnf

```diff
@@ -70,15 +70,16 @@ reserved-char  = %x00-08        ; omit HTAB and LF
                / %x7E-D7FF      ; omit surrogates
                / %xE000-10FFFF
 
-; based on https://www.w3.org/TR/xml/#NT-Name,
-; but cannot start with U+003A COLON ":"
-name = name-start *name-char
+; based on https://www.w3.org/TR/REC-xml-names/#NT-QName
+name       = [namespace ":"] name-body
+namespace  = name-part
+name-part  = name-start *name-char
 name-start = ALPHA / "_"
            / %xC0-D6 / %xD8-F6 / %xF8-2FF
            / %x370-37D / %x37F-1FFF / %x200C-200D
            / %x2070-218F / %x2C00-2FEF / %x3001-D7FF
            / %xF900-FDCF / %xFDF0-FFFD / %x10000-EFFFF
-name-char  = name-start / DIGIT / "-" / "." / ":"
+name-char  = name-start / DIGIT / "-" / "."
            / %xB7 / %x300-36F / %x203F-2040
```

Comment:
This also affects `unquoted`, invalidating e.g. `{2023-11-15T17:23:42Z :datetime}` (which has been cited as desirable to some IIRC). But we're expecting unquoted literals to include strings that won't be valid variable/function/option names anyway (such as `-40`). So probably _this_ PR should get to a place like
```diff
-variable = "$" name
+variable = "$" name-part
 function = (":" / "+" / "-") name
 option = name [s] "=" [s] (literal / variable)
 …
 ; based on https://www.w3.org/TR/xml/#NT-Nmtoken,
 ; but cannot start with U+002D HYPHEN-MINUS or U+003A COLON ":"
-unquoted = unquoted-start *name-char
+unquoted = unquoted-start *unquoted-char
-unquoted-start = name-start / DIGIT / "."
+unquoted-start = name-part-start / DIGIT / "."
                / %xB7 / %x300-36F / %x203F-2040
+unquoted-char  = name-part-char / ":"
 …
-; based on https://www.w3.org/TR/xml/#NT-Name,
-; but cannot start with U+003A COLON ":"
-name = name-start *name-char
-name-start = ALPHA / "_"
-           / %xC0-D6 / %xD8-F6 / %xF8-2FF
-           / %x370-37D / %x37F-1FFF / %x200C-200D
-           / %x2070-218F / %x2C00-2FEF / %x3001-D7FF
-           / %xF900-FDCF / %xFDF0-FFFD / %x10000-EFFFF
-name-char  = name-start / DIGIT / "-" / "." / ":"
-           / %xB7 / %x300-36F / %x203F-2040
+; matches https://www.w3.org/TR/REC-xml-names/#NT-QName
+name            = [namespace ":"] name-part
+namespace       = name-part
+; matches https://www.w3.org/TR/REC-xml-names/#NT-NCName
+name-part       = name-part-start *name-part-char
+name-part-start = ALPHA / "_"
+                / %xC0-D6 / %xD8-F6 / %xF8-2FF
+                / %x370-37D / %x37F-1FFF / %x200C-200D
+                / %x2070-218F / %x2C00-2FEF / %x3001-D7FF
+                / %xF900-FDCF / %xFDF0-FFFD / %x10000-EFFFF
+name-part-char  = name-part-start / DIGIT / "-" / "."
+                / %xB7 / %x300-36F / %x203F-2040
```